appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/processor/ApplicationProcessor.java

```diff
@@ -1087,8 +1075,8 @@ private boolean insertObjectReference(ApiContext context, Reference<?> referee,
                     // Create the schema
                     if (context.isAllowedType(referenceClassType)) {
                         visitSchema(schemaAnnotation, referenceClass, context);
-                    } else {
-                        apiWalker.processAnnotations(singleton(referenceClassType), Schema.class, this::visitSchema);
+                    } else if(referenceClassType instanceof ClassModel) {
+                        apiWalker.processAnnotation((ClassModel)referenceClassType, this);
```

Comment:
Makes me wonder if there is an `else` case now - is it an error? Should we maybe log something?

Reply:
This comes from PR (https://github.com/payara/Payara/pull/4764) although we now support scanning for the packaged jar that's an optional feature.

Reply:
I will add log statement, thx.